src/message_filter.rs

```diff
@@ -36,34 +36,39 @@ impl FilteringResult {
 
 // Structure to filter (throttle) incoming and outgoing messages.
 pub(crate) struct MessageFilter {
-    incoming: LruCache<MessageId, ()>,
-    outgoing: LruCache<(MessageId, XorName), ()>,
+    incoming: Cache<MessageId, ()>,
+    outgoing: Cache<(MessageId, XorName), ()>,
 }
 
 impl MessageFilter {
     pub fn new() -> Self {
         Self {
-            incoming: LruCache::with_expiry_duration_and_capacity(
-                INCOMING_EXPIRY_DURATION,
-                MAX_ENTRIES,
+            incoming: Cache::new(
+                Some(INCOMING_EXPIRY_DURATION),
```

Comment:
better as to the LruCache, avoid passing Option and make the constructor name more explicit

Reply:
Will be addressing all cache things (still wip) in follow up PR.